### PR TITLE
Configure Travis CI to run pre-commit checks

### PR DESCRIPTION
**Pull request by klochenok**
Add travis config file which defines programming language and version the project uses, scripts and commands which should be executed before build.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  hooks:
+  - id: trailing-whitespace
+  - id: flake8
+  - id: check-merge-conflict
+  - id: double-quote-string-fixer
+  - id: end-of-file-fixer
+  - id: name-tests-test
+    include: cherrypy/test/
+    args:
+    - '--django'
```
Comment (webknjaz) [inline .pre-commit-config.yaml]: Please remove quotes

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  - id: check-ast
+  - id: check-byte-order-marker
+  - id: check-case-conflict
+  - id: check-docstring-first
+  - id: check-json
+  - id: pretty-format-json
+  - id: check-symlinks
+  - id: check-yaml
+  - id: detect-private-key
+
```
Comment (webknjaz) [inline .pre-commit-config.yaml]: Please remove one line.

File: mysite/local_settings.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import os
```
Comment (webknjaz) [inline mysite/local_settings.py]: This file is committed by mistake. `git rm --cached mysite/local_settings.py; echo 'local_settings.py' >> .gitignore; git add .gitignore` would fix this.

File: mysite/settings.py [hunk 4]
```diff
@@ -122,4 +122,23 @@
 # https://docs.djangoproject.com/en/1.9/howto/static-files/
 
 STATIC_URL = '/static/'
-STATIC_ROOT = os.path.join(BASE_DIR, 'static')
\ No newline at end of file
+<<<<<<< HEAD
```
Comment (webknjaz) [inline mysite/settings.py]: Please learn about merge conflict markers and fix this module accordingly. (You have to pick what to keep in this module and what to wipe out).
Also please run `pre-commit install` to enable pre-commit tool do checks automatically in the local repo.

**Review by webknjaz — CHANGES_REQUESTED** [review]
According to this branch name you are integrating the TravisCI checks into the repo. So please:
1) [X] Adjust PR title and description accrodingly
2) [X] Go to travis-ci.org and enable this repo there. (Hint: you may want to tick a checkbox in settings, so it would run build only if there's `.travis.yml` in the repo)
3) [X] Add `.travis.yml` to this branch.

File: .gitignore [hunk 5]
```diff
@@ -3,4 +3,4 @@ __pycache__
 girlsenv
 db.sqlite3
 /static
-.DS_Store
\ No newline at end of file
+.DS_Storelocal_settings.py
```
Comment (webknjaz) [inline .gitignore]: It's a typo which has to be separated.

File: mysite/settings.py (hunk 4 above)
Comment (webknjaz) [reply mysite/settings.py]: this is still not fixed

**Review by webknjaz — CHANGES_REQUESTED** [review]
Please fix all issues I stated. Also it would be better to name this PR `Configure Travis CI to run pre-commit checks`

**Comment by webknjaz** [comment]
@klochenok any update?

File: mysite/settings.py [hunk 6]
```diff
@@ -116,29 +116,8 @@
...
-DATABASES['default'] = dj_database_url.config()
-
-SECURE_PROXY_SSL_HEADER = ('HTTP_X_FORWARDED_PROTO', 'https')
-
-ALLOWED_HOSTS = ['*']
-
-DEBUG = False
-
-try:
-    from .local_settings import *
```
Comment (webknjaz) [inline mysite/settings.py]: Why did you remove this version instead of leaving it here?

Comment (klochenok) [reply mysite/settings.py]: I think that it's a solution of merge conflict.

Comment (webknjaz) [reply mysite/settings.py]: It is one of the solutions for the merge. But is it really what you need? According to the tutorial, you the code you removed.

**Review by webknjaz — APPROVED** [review]

**Pull request merged**